// react-response/lib/utils/ReactResponseGreeter.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ReactResponseGreeter = function (_React$Component) {
    _inherits(ReactResponseGreeter, _React$Component);

    function ReactResponseGreeter() {
        _classCallCheck(this, ReactResponseGreeter);

        return _possibleConstructorReturn(this, Object.getPrototypeOf(ReactResponseGreeter).apply(this, arguments));
    }

    _createClass(ReactResponseGreeter, [{
        key: 'render',
        value: function render() {

            return _react2.default.createElement(
                'div',
                { style: {
                        padding: '2em',
                        width: '50%',
                        maxWidth: '600px'
                    } },
                _react2.default.createElement(
                    'h1',
                    { style: {
                            fontFamily: 'Arial',
                            fontSize: '5em',
                            color: '#00BFFF'
                        } },
                    'Hello world from React Response!'
                ),
                _react2.default.createElement(
                    'p',
                    null,
                    'You\'re almost there! To render your own app instead of this Hello World page, pass the root component of your app as a child to the Response component.'
                )
            );
        }
    }]);

    return ReactResponseGreeter;
}(_react2.default.Component);

exports.default = ReactResponseGreeter;
module.exports = exports['default'];
