// react-response/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _ReactServer = require('./ReactServer');

var _ReactServer2 = _interopRequireDefault(_ReactServer);

var _Response = require('./Response');

var _Response2 = _interopRequireDefault(_Response);

var _Route = require('./Route');

var _Route2 = _interopRequireDefault(_Route);

var _server = require('./server');

var _middleware = require('./middleware');

var _reactRouterResponse = require('./handlers/reactRouterResponse');

var _simpleResponse = require('./handlers/simpleResponse');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
    ReactServer: _ReactServer2.default,
    Response: _Response2.default,
    Route: _Route2.default,
    createServer: _server.createServer,
    serve: _server.serve,
    Middleware: _middleware.Middleware,
    Favicon: _middleware.Favicon,
    Static: _middleware.Static,
    createReactRouterResponse: _reactRouterResponse.createReactRouterResponse,
    createSimpleResponse: _simpleResponse.createSimpleResponse
};
module.exports = exports['default'];
