// react-response/lib/Route.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var factory = function factory() {

    var buildServer = function buildServer(props) {
        return {
            route: {
                path: props.path,
                method: props.method
            },
            children: props.children
        };
    };

    var Route = function (_Component) {
        _inherits(Route, _Component);

        function Route() {
            _classCallCheck(this, Route);

            return _possibleConstructorReturn(this, Object.getPrototypeOf(Route).apply(this, arguments));
        }

        return Route;
    }(_react.Component);

    Route.propTypes = {
        path: _react.PropTypes.string.isRequired,
        method: _react.PropTypes.string.isRequired
    };
    Route.defaultProps = {
        path: "/",
        method: "get"
    };
    Route.buildServer = buildServer;


    return Route;
};

exports.default = factory();
module.exports = exports['default'];
